python/fledge/common/logger.py

```diff
@@ -181,10 +202,37 @@ def get_logger(self, logger_name: str):
             logger: returns logger for module
         """
         _logger = logging.getLogger(logger_name)
+
+        # save the old logging.error function
+        __logging_error = _logger.error
+
         console_handler = self.get_console_handler()
         syslog_handler = self.get_syslog_handler()
         self.add_handlers(_logger, [syslog_handler, console_handler])
         _logger.propagate = False
+
+        @wraps(_logger.error)
+        def error(msg, *args, **kwargs):
+            """Case: When we pass exception in error and having different args
+            For example:
+            a) _logger.error(ex)
+            b) _logger.error(ex, "Failed to add data.")
+            """
+            if isinstance(msg, Exception):
+                trace_msg = traceback.format_exception(msg.__class__, msg, msg.__traceback__)
+                if args:
+                    trace_msg[:0] = ["{}\n".format(args[0])]
+                [__logging_error(line.strip('\n')) for line in trace_msg]
+            else:
+                """Case: When we pass string in error
+                For example:  
+                a) _logger.error(str(ex))
+                b) _logger.error("Failed to add data for key: {} along with error:{}".format("Test", str(ex)))
```

Comment:
a) _logger.error("Failed to log audit trail entry")
           b) _logger.error("Failed to log audit trail entry '{}' \n{}".format(code, str(ex))) # \n will result the str(ex) to be printed in next line